src/Modules/Profile/Commands.Profile/InvokePowerBIRestMethod.cs

```diff
@@ -73,36 +107,95 @@ public override void ExecuteCmdlet()
 
                 HttpResponseMessage response = null;
-                switch (requestType)
+                if (string.IsNullOrEmpty(this.OutFile))
                 {
-                    case PowerBIWebRequestMethod.Get:
-                        response = await client.GetAsync(url);
-                        break;
-                    case PowerBIWebRequestMethod.Post:
-                        response = await client.PostAsync(url, new StringContent(body));
-                        break;
-                    case PowerBIWebRequestMethod.Delete:
-                        response = await client.DeleteAsync(url);
-                        break;
-                    case PowerBIWebRequestMethod.Put:
-                        response = await client.PutAsync(url, new StringContent(body));
-                        break;
-                    case PowerBIWebRequestMethod.Patch:
-                        response = await client.SendAsync(new HttpRequestMessage(new HttpMethod("PATCH"), url) { Content = new StringContent(body) });
-                        break;
-                    case PowerBIWebRequestMethod.Options:
-                        response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Options, url));
-                        break;
-                    default:
-                        throw new NotSupportedException($"{nameof(requestType)} of value {requestType} is not supported");
+                    switch (requestType)
+                    {
+                        case PowerBIWebRequestMethod.Get:
+                            response = await client.GetAsync(url);
+                            break;
+                        case PowerBIWebRequestMethod.Post:
+                            response = await client.PostAsync(url, new StringContent(body));
+                            break;
+                        case PowerBIWebRequestMethod.Delete:
+                            response = await client.DeleteAsync(url);
+                            break;
+                        case PowerBIWebRequestMethod.Put:
+                            response = await client.PutAsync(url, new StringContent(body));
+                            break;
+                        case PowerBIWebRequestMethod.Patch:
+                            response = await client.SendAsync(new HttpRequestMessage(new HttpMethod("PATCH"), url) { Content = new StringContent(body) });
+                            break;
+                        case PowerBIWebRequestMethod.Options:
+                            response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Options, url));
+                            break;
+                        default:
+                            throw new NotSupportedException($"{nameof(requestType)} of value {requestType} is not supported");
+                    }
+                }
+                else
+                {
+                    // Stream based, OutFile
```

Comment:
It might be helpful to explain why this is necessary since I know the issue here with the client being disposed was difficult to debug and caused you trouble.

Reply:
Fair point, I will add a comment